Services/Form/classes/class.ilPropertyFormGUI.php

```diff
@@ -481,13 +482,28 @@ function checkInput()
 				}
 			}
 		}
-		
-		
-		if (!$ok && !$this->getDisableStandardMessage())
-		{
-			ilUtil::sendFailure($lng->txt("form_input_not_valid"));
+		$http = $DIC->http();
+		$txt = $DIC->language()->txt("form_input_not_valid");
+		switch ($http->request()->getHeaderLine('Accept')) {
```

Comment:
You control this clientside by means of the XMLHTTPRequest (or what it is called nowadays) or some wrapper, right?

Reply:
Right, by sending a Accept:application/json-Header in your XMLHTTPRequest

Reply:
The changes here in ilPropertyFormGUI could be dropped if too cruel. By sending a ilUtil::sendFailure the message can be rendered with the changes in ilTemplate. But it's more explicit here.